src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDesignTimeDescriptor.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Razor.TagHelpers
+{
+    /// <summary>
+    /// A metadata class containing design time information about a tag helper.
+    /// </summary>
+    public class TagHelperDesignTimeDescriptor
+    {
+        /// <summary>
+        /// Instantiates a new instance of <see cref="TagHelperDesignTimeDescriptor"/>.
+        /// </summary>
+        /// <param name="summary">A summary on how to use a tag helper.</param>
+        /// <param name="remarks">Remarks on how to use a tag helper.</param>
+        /// <param name="outputElementHint">The HTML element a tag helper may output.</param>
+        public TagHelperDesignTimeDescriptor(string summary, string remarks, string outputElementHint)
```

Comment:
We moved toward constructor parameters to make the descriptors immutable while allowing serialization / deserialization. Comparisons, hash codes, and so on get messy if the descriptors can change. I know _we_ wouldn't change the descriptors but we do store some of them in `HashSet`s and so on.

So fine to open an issue but I'm not sure we should ever fix it :smiling_imp: